src/TConsole.cpp

```diff
@@ -30,6 +30,7 @@
 #include "TDockWidget.h"
 #include "TEvent.h"
 #include "TLabel.h"
+#include "TMainConsole.h"
```

Comment:
🤔 That looks dodgy, `TMainConsole` is, I thought, to be derived FROM `TConsole` - so for the base class to include the header for the derived class doesn't seem like a good idea... are you SURE you need this?

Reply:
Yes. I know. Given further refactoring we should be able to get rid of it again, and I have some more changes in mind to get there, but we're not there yet.

I don't want to burden this initial separation with a heap of additional changes that would obfuscate what this patch is doing, which is "as little as possible to get to a viable place we can work further with".

Put another way, it's very helpful to keep `git blame` output in a useable state. Moving lines around *and* modifying them fouls up its moved-lines detection. Squishing patches doesn't help. Thus my desire to do as little as possible here; ideally this last commit shouldn't even be merged into the previous ones.